data/transactions/transaction.go

```diff
@@ -570,6 +571,34 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 		if !proto.Heartbeat {
 			return fmt.Errorf("heartbeat transaction not supported")
 		}
+
+		// If this is a free/cheap heartbeat, it must be very simple.
+		if tx.Fee.Raw < proto.MinTxnFee && tx.Group.IsZero() {
+			kind := "free"
+			if tx.Fee.Raw > 0 {
+				kind = "cheap"
+			}
+
+			if len(tx.Note) > 0 {
```

Comment:
NIT: should we say 'free' heartbeat is not allowed to have these things?

Reply:
We say free if they paid 0, cheap if they pay less than min-fee.